test/cpp/end2end/client_lb_end2end_test.cc

```diff
@@ -2328,6 +2333,11 @@ TEST_F(ClientLbInterceptTrailingMetadataTest,
     auto stream = stub->BidiStream(&ctx);
     ctx.TryCancel();
   }
+  {
+    // Wait for stream to be cancelled.
+    grpc::internal::MutexLock lock(&mu_);
+    cond_.WaitWithTimeout(&mu_, absl::Seconds(3));
```

Comment:
condition variables can wake spuriously so it's best to add a loop with a variable that becomes signalled:

```
while (!trailers_intercepted()) {
  cond_.WaitWithTimeout(&mu_, absl::Seconds(3));
}
```

Reply:
Ah I saw this pattern in some other places and was wondering why it was needed - done